src/freertos_drivers/esp_idf/EspIdfWiFi.cxx

```diff
@@ -0,0 +1,1355 @@
+            mdnsAdvInhibitStaActive_ = false;
+        }
+        return sleep_and_call(
+            &timer_, MDNS_QUERY_INACTIVE_TIMEOUT_NSEC, STATE(mdns_start));
+    }
+}
+
+//
+// EspIdfWiFi::mdns_adv_inhibit()
+//
+void EspIdfWiFi::mdns_adv_inhibit()
+{
+    OSMutexLock locker(this);
+    if (!mdnsAdvInhibit_)
+    {
+        mdnsAdvInhibit_ = true;
+        for (auto it = mdnsServices_.begin(); it != mdnsServices_.end(); ++it)
+        {
+            LOG(VERBOSE, "wifi: mDNS inhibit, service: %s, proto: %s",
+                (*it).name_.c_str(), (*it).proto_.c_str());
+            ::mdns_service_remove((*it).name_.c_str(), (*it).proto_.c_str());
+        }
+    }
+}
+
+//
+// EspIdfWiFi::mdns_adv_inhibit_remove()
+//
+void EspIdfWiFi::mdns_adv_inhibit_remove()
+{
+   OSMutexLock locker(this);
+    if (mdnsAdvInhibit_)
+    {
+        mdnsAdvInhibit_ = false;
+        for (auto it = mdnsServices_.begin(); it != mdnsServices_.end(); ++it)
+        {
+            LOG(VERBOSE, "wifi: mDNS inhibit removed, service: %s, proto: %s",
+                (*it).name_.c_str(), (*it).proto_.c_str());
+            ::mdns_service_add(nullptr, (*it).name_.c_str(),
+                (*it).proto_.c_str(), (*it).port_, nullptr, 0);
+        }
+    }
+}
+
+//
+// EspIdfWiFi::mdns_disable_sta()
+//
+void EspIdfWiFi::mdns_disable_sta()
+{
+#if !defined(CONFIG_MDNS_PREDEF_NETIF_STA)
+    OSMutexLock locker(this);
+    HASSERT(mdnsStaLockCount_ < 10); // Check for runaway asymmetry.
+    if (mdnsStaLockCount_++ == 0)
+    {
+        ESP_ERROR_CHECK(mdns_unregister_netif(staIface_));
+        LOG(VERBOSE, "wifi: STA unregistered on mDNS.");
+    }
+#endif
+}
+
+//
+// EspIdfWiFi::try_restore_sta()
+//
+void EspIdfWiFi::mdns_restore_sta()
+{
+#if !defined(CONFIG_MDNS_PREDEF_NETIF_STA)
+    OSMutexLock locker(this);
+    HASSERT(mdnsStaLockCount_ != 0);
+    if (--mdnsStaLockCount_ == 0)
+    {
+        if (staIface_)
+        {
+            // If the STA interface is not registered by default, register it.
+            ESP_ERROR_CHECK(mdns_register_netif(staIface_));
+            mdns_event_actions_t action = static_cast<mdns_event_actions_t>(
+                MDNS_EVENT_ENABLE_IP4 | MDNS_EVENT_ENABLE_IP6 |
+                MDNS_EVENT_ANNOUNCE_IP4 | MDNS_EVENT_ANNOUNCE_IP6);
+            ESP_ERROR_CHECK(mdns_netif_action(staIface_, action));
+            LOG(VERBOSE, "wifi: STA registered on mDNS.");
+        }
+    }
+#endif
+}
+
+//
+// EspIdfWiFi::wifi_event_handler()
+//
+void EspIdfWiFi::wifi_event_handler(esp_event_base_t base, int32_t id, void *data)
+{
+    if (base != WIFI_EVENT)
+    {
+        LOG(WARNING, "wifi: wifi_event_handler(), not a WIFI_EVENT.");
+        return;
+    }
+
+    switch (id)
+    {
+        default:
+            LOG(INFO, "wifi: wifi_event_handler() unknown id: %i", (int)id);
+            break;
+        case WIFI_EVENT_WIFI_READY:
+            started_ = true;
+            LOG(INFO, "wifi: WiFi started.");
```

Comment:
consider log level adjustments throughout this file, we should likely only report connect / disconnect events specifically by default